modules/codegen/src/smithy4s/codegen/internals/ModelLoader.scala

```diff
@@ -45,12 +48,23 @@ private[codegen] object ModelLoader {
     val currentClassLoader = this.getClass().getClassLoader()
     val deps = resolveDependencies(dependencies, localJars, repositories)
 
-    val modelsInJars = deps.flatMap { files =>
-      val manifestUrl =
-        ModelDiscovery.createSmithyJarManifestUrl(files.getAbsolutePath())
-      try { ModelDiscovery.findModels(manifestUrl).asScala }
-      catch {
-        case _: ModelManifestException => Seq.empty
+    val modelsInJars = deps.flatMap { file =>
+      Using.resource(
+        // Note: On JDK13+, the second parameter is redundant.
+        FileSystems.newFileSystem(file.toPath(), null)
+      ) { jarFS =>
+        val p = jarFS.getPath("META-INF", "smithy", "manifest")
+
+        if (Files.exists(p)) {
+          try ModelDiscovery.findModels(p.toUri().toURL()).asScala.toList
```

Comment:
I have some trust in this approach, as it avoids dealing with any String-based path/uri manipulation, unlike `ModelDiscovery.createSmithyJarManifestUrl` which takes a String.